src/cdef.rs

```diff
@@ -401,11 +402,76 @@ pub fn cdef_padded_tile_copy<T: Pixel>(
   out
 }
 
+// Allocates and returns a new Frame with its own memory that is
+// padded with the input frame
+pub fn cdef_padded_frame_copy<T: Pixel>(in_frame: &Frame<T>) -> Frame<u16> {
```

Comment:
Is there a reason not to write this in terms of or just use cdef_padded_tile_copy()?  The frame can always be wrapped on the fly using Frame::as_tile()

Reply:
This is for performance reason. I can replace it to cdef_padded_tile_copy() and the output is same. But the speed is little bit slow. Therefore, I make a function that do padded process as in original cdef_filter_frame().

Reply:
Ah.  Mid-term, padding goes away, so both will get removed.  I'm surprised as_tile() has that much of a performance impact (and I am using as_tile() extensively elsewhere, so now I'm intrigued as to why).

In any case, as this is temporary and there is a reason (maybe not a good one) I have no serious objection.

Reply:
I have not checked which one of as_tile() or cdef_padded_tile_copy() has performance impact. For now, current approach solves this.